lib/app/features/config/providers/force_update_provider.c.dart

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ion/app/features/config/providers/config_provider.c.dart';
+import 'package:ion/app/features/config/providers/force_update_last_sync_date_provider.c.dart';
+import 'package:ion/app/features/core/providers/app_lifecycle_provider.c.dart';
+import 'package:ion/app/utils/force_update.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'force_update_provider.c.freezed.dart';
+part 'force_update_provider.c.g.dart';
+
+@freezed
+class ForceUpdateState with _$ForceUpdateState {
+  const factory ForceUpdateState({
+    @Default(false) bool showUpdateModal,
+  }) = _ForceUpdateState;
+
+  const ForceUpdateState._();
+}
+
+@Riverpod(keepAlive: true)
+class ForceUpdate extends _$ForceUpdate {
+  static const int eightHoursInMilliseconds = 8 * 60 * 60 * 1000;
+
+  @override
+  ForceUpdateState build() {
+    _checkAndUpdateConfig();
+
+    ref.listen<AppLifecycleState>(
+      appLifecycleProvider,
+      (previous, next) {
+        if (next == AppLifecycleState.resumed) {
+          _checkAndUpdateConfig();
```

Comment:
Let's make a method `_checkAndUpdateConfig` public and call it in the `AppLifecycleObserver`class